bin/start-codepipeline/index.ts

```diff
@@ -0,0 +1,268 @@
+    return await new Promise<AWS.CodePipeline.StartPipelineExecutionOutput>(
+      (resolve, reject) => {
+        this.codepipeline.startPipelineExecution(
+          {
+            name: this.pipelineName,
+          },
+          function (
+            err: AWS.AWSError | undefined,
+            okData: AWS.CodePipeline.StartPipelineExecutionOutput
+          ) {
+            if (err) {
+              console.error(err);
+              reject(err);
+            } else {
+              resolve(okData);
+            }
+          }
+        );
+      }
+    );
+  }
+
+  async updatePipeline(
+    existingPipeline?: AWS.CodePipeline.PipelineDeclaration
+  ) {
+    return await new Promise<boolean>((resolve, reject) => {
+      if (!existingPipeline) {
+        reject(
+          new Error(
+            `Pipeline '${this.pipelineName}' has no stages or does not exist`
+          )
+        );
+      }
+
+      if (!this.gitBranch) {
+        resolve(true);
+        return;
+      }
+
+      const pipeline = existingPipeline!;
+      const sourceAction = pipeline.stages[0]!.actions[0]!;
+      sourceAction.configuration!.Branch = this.gitBranch;
+
+      this.codepipeline.updatePipeline(
+        {
+          pipeline: {
+            ...pipeline,
+            stages: [...pipeline.stages],
+          },
+        },
+        function (
+          err: AWS.AWSError | undefined,
+          okData: AWS.CodePipeline.UpdatePipelineOutput
+        ) {
+          if (err) {
+            console.error(err);
+            reject(err);
+          } else {
+            console.log("Pipeline's been updated with the new source");
+            resolve(!!okData);
+          }
+        }
+      );
+    });
+  }
+
+  async getPipeline() {
+    return await new Promise<AWS.CodePipeline.PipelineDeclaration>(
+      (resolve, reject) => {
+        this.codepipeline.getPipeline(
+          {
+            name: this.pipelineName,
+          },
+          function (
+            err: AWS.AWSError | undefined,
+            okData: AWS.CodePipeline.GetPipelineOutput
+          ) {
+            if (err) {
+              console.error(err);
+              reject(err);
+            } else {
+              resolve(okData.pipeline!);
+            }
+          }
+        );
+      }
+    );
+  }
+
+  async getPipelineExecution(executionId: string) {
+    return await new Promise<AWS.CodePipeline.PipelineExecution>(
+      (resolve, reject) => {
+        this.codepipeline.getPipelineExecution(
+          {
+            pipelineName: this.pipelineName,
+            pipelineExecutionId: executionId,
+          },
+          function (
+            err: AWS.AWSError | undefined,
+            okData: AWS.CodePipeline.GetPipelineExecutionOutput
+          ) {
+            if (err) {
+              console.error(err);
+              reject(err);
+            } else {
+              resolve(okData.pipelineExecution!);
+            }
+          }
+        );
+      }
+    );
```

Comment:
```ts
try {
  const result = await this.codepipeline.getPipelineExecution({
    pipelineName: this.pipelineName,
    pipelineExecutionId: executionId,
  }).promise();
  return result.pipelineExecution!;
} catch (e) {
  console.error(e); // although this could be moved up higher.
  throw e;
}
```